src/main/java/Parser.java

```diff
@@ -0,0 +1,92 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class Parser {
+
+    public Parser() {}
+
+    public void executeCommand(String command) throws InvalidCommandException, MissingInputException,
+            MissingDescriptionException {
+        String[] commandWords = command.trim().split(" ");
+        String commandType = commandWords[0];
+        TaskList taskList = new TaskList();
+        switch (commandType) {
+            case "list":
+                taskList.printList();
+                break;
+            case "done":
+                int taskNumber = Integer.parseInt(commandWords[1]);
+                taskList.markAsDone(taskNumber);
+                break;
+            case "delete":
+                int taskNumber2 = Integer.parseInt(commandWords[1]);
+                taskList.deleteTask(taskNumber2);
+                break;
+            case "todo":
+            case "deadline":
+            case "event":
+                addTask(command, commandType);
+                break;
+            case "bye":
+                Ui ui = new Ui();
```

Comment:
This line of code causes the execution of Duke to create 2 UIs, one on start of Duke and one when "bye" is called. The code should work but I feel like it would cause a bit of confusion in the future, especially when you add on extensions to the UI. Is there anyway you can pass the execution of exit back to the first UI instead of creating a new one?